packages/components/src/presentation-components/properties/inputs/UniquePropertyValuesSelector.tsx

```diff
@@ -221,27 +198,46 @@ function getFieldClassInfos(field?: Field) {
 
 interface UseUniquePropertyValuesLoaderProps {
   imodel: IModelConnection;
+  property: PropertyDescription;
+  descriptor: Descriptor;
   ruleset?: Ruleset;
   field?: Field;
   descriptorInputKeys?: Keys;
 }
 
-function useUniquePropertyValuesLoader({ imodel, ruleset, field, descriptorInputKeys }: UseUniquePropertyValuesLoaderProps) {
+function useUniquePropertyValuesLoader({ imodel, property, descriptor, ruleset, field, descriptorInputKeys }: UseUniquePropertyValuesLoaderProps) {
+  const [loadedCount, setLoadedCount] = useState(0);
```

Comment:
Do we need `loadedCount`? Isn't `loadedOptions` enough?

Reply:
`loadedCount` and `loadedOptions.length` can differ. The `loadedCount`  is used for determining the starting index for the next page and it keeps track of the total amount of values regardless of their content while `loadedOptions` does not include values with missing properties (for example, with no `displayValue`).

Reply:
Ok. I still think all three pieces of state should be merged into one.

Reply:
I agree, merged into one.